dali/test/python/operator_2/test_random_choice.py

```diff
@@ -0,0 +1,201 @@
+        What kind of `a` parameter to test: "scalar" - just a scalar value, "0d" - list of scalars
+    elem_shape : tuple of int
+        Shape of the elements sampled by the distribution, non-empty tuple allowed only for "nd"
+        case.
+    use_p : bool
+        Pass the p parameter or use uniform distribution.
+    output_shape : tuple
+        Parameter requesting the shape of output, the resulting shape will be
+        `output_shape + elem_shape`.
+    """
+
+    if kind in {"scalar", "0d"}:
+        assert elem_shape == (), "elem_shape must be empty tuple for scalar and 0d case"
+    sampling_offset = 5  # offset compared to the index in the batch
+    batch_size = 8
+    if output_shape is None or output_shape == ():
+        n_iters = 1000
+    else:
+        n_iters = 5
+
+    def get_p(n):
+        """
+        Return probabilities for n elements, i-th element is proportional to its index in batch.
+        """
+        p = np.zeros((n,), dtype=np.float32)
+        for i in range(n):
+            p[i] = (i + 1) / (n * (n + 1) / 2)
+        return p
+
+    def inp(sample_info):
+        n = sample_info.idx_in_batch + sampling_offset
+        p = get_p(n)
+        if kind == "scalar":
+            return np.array(n), p
+        else:  # "0d"
+            return np.arange(sampling_offset, n + sampling_offset), p
+
+    @pipeline_def(batch_size=batch_size, device_id=0, num_threads=4, seed=1234)
+    def choice_pipe():
+        a, p = fn.external_source(inp, batch=False, num_outputs=2)
+        if not shape_like:
+            choice = fn.random.choice(a, p=p if use_p else None, shape=output_shape)
+        else:
+            choice = fn.random.choice(a, np.full(output_shape, 42), p=p if use_p else None)
+        return choice, a, p
+
+    tuple_output_shape = output_shape if output_shape is not None else ()
+
+    pipe = choice_pipe()
+    pipe.build()
+    choices = [[] for _ in range(batch_size)]
+    for _ in range(n_iters):
+        ch, a, p = pipe.run()
+        for i in range(batch_size):
+            assert tuple(ch[i].shape()) == tuple_output_shape + elem_shape
+            # Extract and accumulate values in samplewise fashion
+            choices[i].append(np.array(ch[i]))
+    for i in range(batch_size):
+        choices[i] = np.stack(choices[i])
+        check_sample(
+            choices[i],
+            size=(n_iters,) + tuple_output_shape,
+            a=np.array(a[i]),
+            p=np.array(p[i]) if use_p else None,
+            idx=i,
+        )
+
+
+@params(
+    *[
+        (
+            (1,),
+            {"p": 1.5},
+            "Probabilities must be in range *, but got: 1.5 for sample: 0 at index 0.",
+        ),
+        (
+            (2,),
+            {"p": 0.25},
+            "Sum of probabilities must be 1.0, but got 0.5 for sample: 0.",
+        ),
+        (
+            (-5,),
+            {},
+            "Expected positive number of elements for sampling, got: -5 for sample: 0.",
+        ),
```

Comment:
How about adding a test with 0 elements? It's less obvious than -5

Reply:
done